RELEASING.md

```diff
@@ -0,0 +1,23 @@
+# How To Make A New Release Of PyGLS
+
+1. Update the version number in `pyproject.toml`
+
+-   Try to follow https://semver.org/
+-   Commit message title should be something like: `build: v1.3.0`
+-   Typically you'll want to make a dedicated PR for the version bump. But if a previous PR has already merged the wanted version bump into main, then a new PR is not necessary.
+-   Example PR for a release https://github.com/openlawlibrary/pygls/pull/434
+-   Merge the PR
+
+2. Create a Github Release
+
+-   Goto https://github.com/openlawlibrary/pygls/releases/new
+-   In the "Choose a tag" dropdown button, type the new version number.
+-   Click the "Generate release notes" button.
+-   Generating release notes does not require making an actual release, so you may like to copy and paste the output of the release notes to make the above release PR more informative.
+-   Click the green "Publish release button".
```

Comment:
Maybe add a note to check the "Set as a pre-release" button if required.

Reply:
Thanks, done ☑️